docs/api/lib-app.adoc

```diff
@@ -0,0 +1,441 @@
+= Application library
+:toc: right
+:imagesdir: ../images
+
+Functions to fetch and manipulate applications.
+
+== Usage
+
+Add the following into your `build.gradle` file:
+
+[source,groovy]
+----
+dependencies {
+  include "com.enonic.xp:lib-app:${xpVersion}"
+}
+----
+
+In your JavaScript controller, add a require statement:
+
+```js
+var appLib = require('/lib/xp/app');
+```
+
+You are now free to use the library functionality.
+
+== Functions
+
+=== get
+
+Returns installed application. If both local and virtual apps are installed with specified key, then local will be in the result.
+
+[.lead]
+Parameters
+
+[%header,cols="1%,1%,97%a"]
+[frame="none"]
+[grid="none"]
+|===
+| Name   | Type   | Description
+| params | object | JSON with parameters
+
+[%header,cols="1%,1%,97%a"]
+[frame="topbot"]
+[grid="none"]
+[caption=""]
+.Properties
+!===
+! Name  ! Type   !  Description
+! key  ! string !  Application key
+!===
+
+|===
+
+[.lead]
+Returns
+
+*object* : (<<#Application,`Application`>>) Detail information for the application. Or null if the application could not be found.
+
+[.lead]
+Example
+
+.Obtains details for an application:
+```js
+var application = appLib.get('my-app-key');
+
+if (application) {
+  log.info('Application name = %s', application.displayName);
+} else {
+  log.info('Application not found');
+}
+```
+.Return value:
+```js
+var expected = {
+    'key': 'my-app',
+    'displayName': 'app display name',
+    'vendorName': 'vendor name',
+    'vendorUrl': 'https://vendor.url',
+    'url': 'https://myapp.url',
+    'version': '1.0.0',
+    'systemVersion': '4.2.3-SNAPSHOT',
+    'minSystemVersion': '2.0.0',
+    'maxSystemVersion': '3.0.0',
+    'modifiedTime': '2020-09-25T10:00:00Z',
+    'started': true,
+    'system': false
+};
+```
+
+=== list
+
+Fetches all installed applications. If both local and virtual apps are installed with the same key, then local will be in the result.
```

Comment:
Fetches all installed applications. If a local and a virtual app are both installed with the same key, then the local one will take precedence and will be returned in the result.